python/src/puml/helpers/requirements_helpers.py

```diff
@@ -0,0 +1,132 @@
+from typing import List, Optional
+
+
+def _get_requirements_defs(reqs: dict) -> List[dict]:
+    """
+    Helper method for extracting and sorting data from a requirement specification
+
+    Args:
+        reqs (dict): A dictionary of requirement definitions
+
+    Returns:
+        A List of dictionaries of organized data for diagram generation
+    """
+    requirements = []
+    if reqs:
+        for req_key in reqs:
+            req = reqs[req_key]
+            id = req.structure["req"]["id"]
+            attributes = req.structure["req"]["attributes"]
+            type = _get_requirement_type(attributes)
+            title = None
+            name = req.structure["req"]["name"]
+            shall = req.structure["req"]["shall"]
+            connected = _get_connected_requirements(req, reqs)
+            requirements.append({
+                "type": type,
+                "title": title,
+                "name": name,
+                "id": id,
+                "shall": shall,
+                "attributes": attributes,
+                "connected": connected,
+            })
+    return requirements
+
+
+def _get_requirement_type(attributes: List[dict]) -> str:
+    """
+    Helper method for extracting the TADI type of a requirement
+
+    Args:
+        attributes (List[dict]): A List of a requirement definitions attributes
+
+    Returns:
+        The value of the TADI attribute
+    """
+    if not attributes:
+        attributes = [{}]
+    for attribute in attributes:
+        if attribute["name"] == "TADI":
+            print(attribute["value"])
+            return attribute["value"]
+
+
+def _get_connected_requirements(req: dict, reqs: dict) -> List[dict]:
+    """
+    Helper method that finds the parent and child requirement definitions for a specified requirement
+
+    Args:
+        req (dict): A requirement defintion
+        reqs (dict): A dictionary containing all requirement definitions
+
+    Returns:
+        A list of parent and child requirement definitions for the specified requirement
+    """
+    connected_reqs = []
+    if not reqs:
+        return connected_reqs
+
+    for struct in reqs:
+        if req != struct:
+            req_id = req.structure["req"]["id"]
+
+            child = _get_child_requirements(req_id, reqs[struct])
+            if child:
+                connected_reqs.append(child)
+
+            parent = _get_parent_requirements(req_id, reqs[struct])
+            if parent:
+                connected_reqs.append(parent)
```

Comment:
Okay, actually seems possibly something in this is going squirrelly, because it is just circling it seems of up to the parent then through all of the children. 

Here is an excerpt.
```
req_id: ADD-1
child: {'child': 'ADD-1', 'parent': 'SM-1', 'arrow': '+--', 'relationship': ''}
connected_reqs: [{'child': 'ADD-1', 'parent': 'SM-1', 'arrow': '+--', 'relationship': ''}]
req_id: SM-1
parent: {'parent': 'SM-1', 'child': 'ADD-1', 'arrow': '+--', 'relationship': ''}
req_id: SM-1
parent: {'parent': 'SM-1', 'child': 'SUB-1', 'arrow': '+--', 'relationship': ''}
connected_reqs: [{'parent': 'SM-1', 'child': 'ADD-1', 'arrow': '+--', 'relationship': ''}, {'parent': 'SM-1', 'child': 'SUB-1', 'arrow': '+--', 'relationship': ''}]
req_id: SUB-1
child: {'child': 'SUB-1', 'parent': 'SM-1', 'arrow': '+--', 'relationship': ''}
connected_reqs: [{'child': 'SUB-1', 'parent': 'SM-1', 'arrow': '+--', 'relationship': ''}]
connected_reqs: []
connected_reqs: []
req_id: CALC-3
parent: {'parent': 'CALC-3', 'child': 'SM-1', 'arrow': '+--', 'relationship': ''}
req_id: CALC-3
parent: {'parent': 'CALC-3', 'child': 'CM-1', 'arrow': '+--', 'relationship': ''}
req_id: CALC-3
parent: {'parent': 'CALC-3', 'child': 'IM-1', 'arrow': '+--', 'relationship': ''}
connected_reqs: [{'parent': 'CALC-3', 'child': 'SM-1', 'arrow': '+--', 'relationship': ''}, {'parent': 'CALC-3', 'child': 'CM-1', 'arrow': '+--', 'relationship': ''}, {'parent': 'CALC-3', 'child': 'IM-1', 'arrow': '+--', 'relationship': ''}]
connected_reqs: []
req_id: CM-1
parent: {'parent': 'CM-1', 'child': 'SQRT-1', 'arrow': '+--', 'relationship': ''}
req_id: CM-1
parent: {'parent': 'CM-1', 'child': 'POW-1', 'arrow': '+--', 'relationship': ''}
connected_reqs: [{'parent': 'CM-1', 'child': 'SQRT-1', 'arrow': '+--', 'relationship': ''}, {'parent': 'CM-1', 'child': 'POW-1', 'arrow': '+--', 'relationship': ''}]
req_id: SQRT-1
child: {'child': 'SQRT-1', 'parent': 'CM-1', 'arrow': '+--', 'relationship': ''}
connected_reqs: [{'child': 'SQRT-1', 'parent': 'CM-1', 'arrow': '+--', 'relationship': ''}]
```

I have print statements after each append for the child and parent checks. And a print statement in the `_get_requirements_defs` following the `connected` to print what is currently seen in that list.

Reply:
Same issue as above with the explicit_start=True flag.